jpa-crud/jpa-crud-java-sbt/jpa-crud-java-sbt-impl/src/test/java/com/example/jpacrud/application/EntityServiceTest.java

```diff
@@ -0,0 +1,347 @@
+package com.example.jpacrud.application;
+
+import akka.NotUsed;
+import com.example.jpacrud.api.CreateEntityRequest;
+import com.example.jpacrud.api.EntityResource;
+import com.example.jpacrud.api.EntityService;
+import com.example.jpacrud.api.UpdateEntityRequest;
+import com.lightbend.lagom.javadsl.api.transport.BadRequest;
+import com.lightbend.lagom.javadsl.api.transport.NotFound;
+import com.lightbend.lagom.javadsl.testkit.ServiceTest.TestServer;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.pcollections.HashTreePSet;
+
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import static akka.NotUsed.notUsed;
+import static com.lightbend.lagom.javadsl.testkit.ServiceTest.defaultSetup;
+import static com.lightbend.lagom.javadsl.testkit.ServiceTest.startServer;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.pcollections.HashTreePSet.singleton;
+
+import static java.util.Arrays.asList;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+@DisplayName("Test EntityService")
+class EntityServiceTest {
+
+    private static TestServer server;
+    private static EntityService service;
+    private static CreateEntityRequest entityRequest = CreateEntityRequest.builder()
+            .property("I'm property")
+            .parts(HashTreePSet.from(asList("part1", "part2")))
+            .build();
+
+    @BeforeAll
+    static void beforeAll() {
+        server = startServer(defaultSetup().withCluster(false));
+        service = server.client(EntityService.class);
+    }
+
+    @AfterAll
+    static void afterAll() {
+        if (server != null) server.stop();
+    }
+
+    static <T> T eventually(CompletionStage<T> stage) throws InterruptedException, ExecutionException, TimeoutException {
+        return stage.toCompletableFuture().get(5, SECONDS);
+    }
+
+    @Nested
+    @DisplayName("creation entities")
+    class CreateEntityTest {
+
+        @Test
+        @SuppressWarnings("unchecked")
+        @DisplayName("correct entity should be create successfully")
+        void shouldCreateEntity() throws InterruptedException, ExecutionException, TimeoutException {
```

Comment:
JUnit 5! Oh yeah!

It's so good to see people adopting and demoing it! +100

Reply:
👍